src/adr/0036-eventbus.md

```diff
@@ -0,0 +1,73 @@
+# 36. Event bus
+
+Date: 2020-05-05
+
+## Status
+
+Accepted (lazy consensus)
+
+## Context
+
+James mailbox uses an event driven architecture. 
+It means every meaningful action on mailboxes or messages triggers an event for any component to react to that event.
+`MailboxListener` allows executing actions upon mailbox events. They could be used for a wide variety of purposes, like 
+enriching mailbox managers features or enabling user notifications upon concurrent mailboxes operations.
+
+Interactions happen via the managers (`RightManager`, `MailboxManager`, `MessageManager`, `MessageIdManager`) which emit an
+event on the EventBus, which will ensure the relevant MailboxListeners will be executed at least once.
+
+`MailboxListener` can be registered in a work queue fashion on the `EventBus`. Each work queue correspond to a given 
+MailboxListener class with the same configuration, identified by their group. Each event is executed at least once
+within a James cluster, errors are retried with an exponential back-off delay. If the execution keeps failing, the event
+ is stored in `DeadLetter` for later reprocessing, triggered via WebAdmin.
+
+Guice products enable the registration of additional mailbox listeners. A user can furthermore define its own 
+mailboxListeners via the use of `extension-jars`.
+
+MailboxListener can also be registered on a notification fashion on the eventBus. A mailbox listener can be registered 
+via a `registrationKey` identifying entities concerned by the event. Upon event emission, the manager will indicate the 
+`registrationKey` this event should be sent to. A mailboxListener will thus only receive the event for the registration 
```

Comment:
Could you give examples of `registrationKey`s because with this generic definition, it doesn't really help the reader to understand what he can expect from that feature?